packages/snjs/lib/services/api/api_service.ts

```diff
@@ -234,6 +236,21 @@ export class SNApiService extends PureService {
     });
   }
 
+  async getAuthMethods({
+    email,
```

Comment:
I would again aim for consistency here rather than creating a hybrid API of DTO and non-DTO. Personally TypeScript has removed the need for DTOs for me. Might make sense for function with an excessive amount of arguments but here seems unnecessary.

Reply:
I don't really see this as DTO (which seems to be a pretty vague concept), but as named arguments (which aren't really supported in JS). They are the better default, even if you have a super-strong static type system, because such system will not help you with positional arguments of the same type next to each other. Hence my habit of always starting with that in JS/TS. Got bitten by this enough times that I like to err on the safe side.

And named arguments are less brittle in terms of cosmetic refactoring -- you can rearrange all you want without fear of breaking something.

For a single argument this is indeed not necessary and I would probably change this before submitting this PR for review.

Reply:
How do you specify if an argument is required or optional with named args? Maybe all required args can be statically typed, and the rest can be via "DTO" or named args?

Reply:
> How do you specify if an argument is required or optional with named args?

Similarly to positional:

```
{
  requiredArg,
  optionalArg = default,
}: {
  requiredArg: Type1
  optionalArg?: Type2,
}
```

> Maybe all required args can be statically typed, and the rest can be via "DTO" or named args?

All should be statically typed unless there is a really good reason not to.

If you mean required should be positional and optional should be named -- yes, that often happens, but it doesn't solve the issues of mixing up order for args of the same type and impractical refactorability -- they are orthogonal to that rule.

***

The major objection to named args is verbosity, especially in TS -- you can't type each arg inline. 

Hairs can also be split in JS about performance.

Other than that, named are just better for humans and you can't make a positional error with them.

That said, the objections are sometimes valid and reason has to be applied. No need to stick named args everywhere if it may improve things.

2 or 3 positional args might be fine, so long as they are of different types. Same types could also be ok if the order doesn't really matter (if the operation is commutative and guaranteed to stay that way -- but again, you have to think this through case-by-case).

Reply:
fixed, so resolving